pkg/controller/servicemesh/webhooks/mutation/controlplane.go

```diff
@@ -299,3 +320,19 @@ func (m *smcpv2mutator) SetOpenShiftRouteEnabled(value bool) {
 
 	m.patches = append(m.patches, jsonpatch.NewPatch("add", "/spec/gateways", *gateways))
 }
+
+func (m *smcpv1mutator) IsTracingTypeSpecified() bool {
+	return false
+}
+
+func (m *smcpv1mutator) SetTracingType(value v2.TracerType) {}
+
+func (m *smcpv2mutator) IsTracingTypeSpecified() bool {
+	tracing := m.smcp.Spec.Tracing
+
+	return !(tracing == nil || tracing.Type == "")
+}
+
+func (m *smcpv2mutator) SetTracingType(value v2.TracerType) {
+	m.patches = append(m.patches, jsonpatch.NewPatch("add", "/spec/tracing/type", string(value)))
```

Comment:
Setting `/spec/tracing/type` will fail when `/spec/tracing` is empty. We need to construct a `TracingConfig`  while preserving any user defined values.